This fixes a build issue caused by linking order.

Upstream-Status: Inappropriate
Most distos have there own verison of this fix. This was derived by
* Fix link order to list libraries after the objects that require them
(LP: #771080).

-- Colin Watson <cjwatson@ubuntu.com>  Tue, 13 Sep 2011 10:07:08 +0100


signed-off-by: Armin Kuster <akuster808@gmail.com>

Index: netkit-rsh-0.17/rsh/Makefile
===================================================================
--- netkit-rsh-0.17.orig/rsh/Makefile
+++ netkit-rsh-0.17/rsh/Makefile
@@ -6,7 +6,7 @@ include ../MRULES
 OBJS = rsh.o
 
 rsh: $(OBJS)
-	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
+	$(CC) $^ -o $@ $(LDFLAGS) $(LIBS)
 
 install: rsh
 	install -o root -m$(SUIDMODE) rsh $(INSTALLROOT)$(BINDIR)
Index: netkit-rsh-0.17/rshd/Makefile
===================================================================
--- netkit-rsh-0.17.orig/rshd/Makefile
+++ netkit-rsh-0.17/rshd/Makefile
@@ -11,7 +11,7 @@ LIBS += -ldl -lpam
 endif
 
 rshd: $(OBJS)
-	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
+	$(CC) $^ -o $@ $(LDFLAGS) $(LIBS)
 
 install: rshd
 	install -m$(DAEMONMODE) rshd $(INSTALLROOT)$(SBINDIR)/in.rshd
Index: netkit-rsh-0.17/rlogin/Makefile
===================================================================
--- netkit-rsh-0.17.orig/rlogin/Makefile
+++ netkit-rsh-0.17/rlogin/Makefile
@@ -7,7 +7,7 @@ PROG=rlogin
 OBJS=rlogin.o
 
 $(PROG): $(OBJS)
-	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
+	$(CC) $^ -o $@ $(LDFLAGS) $(LIBS)
 
 install: $(PROG)
 	install -o root -m$(SUIDMODE) $(PROG) $(INSTALLROOT)$(BINDIR)
Index: netkit-rsh-0.17/rlogind/Makefile
===================================================================
--- netkit-rsh-0.17.orig/rlogind/Makefile
+++ netkit-rsh-0.17/rlogind/Makefile
@@ -13,7 +13,7 @@ LIBS += -ldl -lpam -lpam_misc
 endif
 
 rlogind: $(OBJS)
-	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
+	$(CC) $^ -o $@ $(LDFLAGS) $(LIBS)
 
 rlogind.o: pathnames.h logwtmp.h rlogind.h ../version.h
 logwtmp.o: logwtmp.h
Index: netkit-rsh-0.17/rexecd/Makefile
===================================================================
--- netkit-rsh-0.17.orig/rexecd/Makefile
+++ netkit-rsh-0.17/rexecd/Makefile
@@ -24,7 +24,7 @@ endif
 CFLAGS += -DRESTRICT_FTP=1
 
 rexecd: rexecd.o
-	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
+	$(CC) $^ -o $@ $(LDFLAGS) $(LIBS)
 
 install: rexecd
 	install -m$(DAEMONMODE) rexecd $(INSTALLROOT)$(SBINDIR)/in.rexecd
